maven-core/src/main/java/org/apache/maven/classrealm/DefaultClassRealmManager.java

```diff
@@ -301,20 +300,18 @@ private void callDelegates(
     }
 
     private Set<String> populateRealm(ClassRealm classRealm, List<ClassRealmConstituent> constituents) {
-        Set<String> includedIds = new LinkedHashSet<>();
+        Set<String> includedIds = new LinkedHashSet<>(constituents.size());
 
-        if (logger.isDebugEnabled()) {
-            logger.debug("Populating class realm " + classRealm.getId());
-        }
+        logger.debug("Populating class realm {}", classRealm.getId());
 
         for (ClassRealmConstituent constituent : constituents) {
             File file = constituent.getFile();
 
-            String id = getId(constituent);
-            includedIds.add(id);
-
             if (logger.isDebugEnabled()) {
-                logger.debug("  Included: " + id);
+                String id = getId(constituent);
```

Comment:
Same here.  Move back the `includedIds` computation and simply remove the `isDebugEnabled()` call.